neuralmonkey/decoders/transformer.py

```diff
@@ -529,4 +529,10 @@ def body(*args) -> LoopState:
         # pylint: enable=too-many-locals
 
         return body
+
+    def decoding_loop(self, train_mode: bool, sample: bool = False,
+                      temperature: float = 1) -> LoopState:
+        with tf.control_dependencies([self.decoding_w, self.decoding_b]):
```

Comment:
nemůže tohle dělat Autoregressive? Když autoregressive má decoding_w a decoding_b?

Reply:
to, ze chces buildit decoding_w/b pred tf.while_loop, je specificke pro tformer (protoze jinak pocita train_loop_result). Ale presunuti do Autoregressive by asi nicemu neuskodilo.

Reply:
já bych to tam dal, ostatním autoregresivcům to neuškodí, když se řekne, že se musí nainicializovat výstupní projekce dřív, než se vleze do while loopu..